BooruSharp.Others/Pixiv.cs

```diff
@@ -200,6 +202,60 @@ public override async Task<SearchResult[]> GetLastPostsAsync(params string[] tag
             return ParseSearchResults((JArray)json["illusts"]);
         }
 
+        public async Task<SearchResult[]> GetRankingAsync(RankingMode mode = RankingMode.Daily, int offset = 0, DateTime? date = null)
+        {
+            if (AccessToken == null)
+                throw new AuthentificationRequired();
+            await CheckUpdateTokenAsync();
+            string url = _baseUrl + "/v1/illust/ranking?mode=" + rankingModeValues[mode];
+            if (offset != 0)
+                url += "&offset=" + offset;
+            if (date.HasValue)
+                url += "&date=" + date.Value.ToString("yyyy-MM-dd");
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Add("Authorization", "Bearer " + AccessToken);
+
+            var http = await HttpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+            if (!http.IsSuccessStatusCode)
+                throw new Exception(http.StatusCode.ToString()); //ToDo: Implement exception
+            JToken json = (JToken)JsonConvert.DeserializeObject(await http.Content.ReadAsStringAsync());
+            return ParseSearchResults((JArray)json["illusts"]);
+        }
+
+        public enum RankingMode
+        {
+            Daily = 0,
+            Weekly = 1,
+            Monthly = 2,
+            Popular_among_male_users = 3,
+            Popular_among_female_users = 4,
+            Original = 5,
+            Manga_Rookie = 6,
+            Manga_Daily = 7,
+            R18_Daily = 8,
+            R18_Weekly = 9,
+            R18_Popular_among_male_users = 10,
+            R18_Popular_among_female_users = 11,
+            R18G_Weekly = 12
+        }
+
+        static private Dictionary<RankingMode, string> rankingModeValues = new Dictionary<RankingMode, string>
+        {
+            { RankingMode.Daily, "day" },
+            { RankingMode.Weekly, "week" },
+            { RankingMode.Monthly, "month" },
+            { RankingMode.Popular_among_male_users, "day_male" },
+            { RankingMode.Popular_among_female_users, "day_female" },
+            { RankingMode.Original, "week_original" },
+            { RankingMode.Manga_Rookie, "week_rookie" },
+            { RankingMode.Manga_Daily, "day_manga" },
+            { RankingMode.R18_Daily, "day_r18" },
+            { RankingMode.R18_Weekly, "week_r18" },
+            { RankingMode.R18_Popular_among_male_users, "day_male_r18" },
+            { RankingMode.R18_Popular_among_female_users, "day_female_r18" },
+            { RankingMode.R18G_Weekly, "week_r18g" }
```

Comment:
```suggestion
            { RankingMode.PopularAmongMaleUsers, "day_male" },
            { RankingMode.PopularAmongFemaleUsers, "day_female" },
            { RankingMode.Original, "week_original" },
            { RankingMode.MangaRookie, "week_rookie" },
            { RankingMode.MangaDaily, "day_manga" },
            { RankingMode.R18Daily, "day_r18" },
            { RankingMode.R18Weekly, "week_r18" },
            { RankingMode.R18PopularAmongMaleUsers, "day_male_r18" },
            { RankingMode.R18PopularAmongFemaleUsers, "day_female_r18" },
            { RankingMode.R18GWeekly, "week_r18g" }
```
As per [Microsoft guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers), PascalCase is the preferred way of naming enum values.